crates/aiken-lang/src/parser/expr/block.rs

```diff
@@ -11,7 +11,17 @@ pub fn parser(
     choice((
         sequence
             .clone()
-            .delimited_by(just(Token::LeftBrace), just(Token::RightBrace)),
+            .delimited_by(just(Token::LeftBrace), just(Token::RightBrace))
+            .map_with_span(|e, span| {
+                if matches!(e, UntypedExpr::Assignment { .. }) {
+                    UntypedExpr::Sequence {
+                        location: span,
+                        expressions: vec![e],
+                    }
+                } else {
+                    e
+                }
```

Comment:
Shouldn't this also be done for `(` and `)` delimiters? I am going to add a few more test cases to ensure the behavior is ok 👍

Reply:
I think that's up to debate but I'll leave it as is for now. This effectively prevent from writing a code block delimited by parens which may not be that bad of an idea. So there are some subtle differences in the use of `{`/`}` and `(`/`)`. It might turn out to be confusing but I also have no problem with:

```
let a = { let b = 42 }
```

be _valid from a parser standpoint_ but then rejected by the type-checker (sequence ending in a let binding) but:

```
let a = ( let b = 42 )
```

be an invalid syntax altogether.

Reply:
This is actually confusing 🤔 ...

Reply:
You're right this is what it does. It allows it to be a valid parse and then the typechecker can now properly catch with assert_no_assignment.